Tests/MapboxNavigationTests/WayNameViewTests.swift

```diff
@@ -0,0 +1,104 @@
+import XCTest
+import MapboxDirections
+import TestHelper
+import MapboxMaps
+@testable import MapboxNavigation
+@testable import MapboxCoreNavigation
+
+class WayNameViewTests: TestCase {
+    lazy var wayNameView: WayNameView = {
+        let view: WayNameView = .forAutoLayout()
+        view.containerView.isHidden = true
+        view.containerView.clipsToBounds = true
+        
+        let config = URLSessionConfiguration.default
+        config.protocolClasses = [ImageLoadingURLProtocolSpy.self]
+        view.label.spriteRepository = SpriteRepositoryStub.init()
+        view.label.spriteRepository.sessionConfiguration = config
+        return view
+    }()
+    
+    override func setUp() {
+        super.setUp()
+        continueAfterFailure = false
+
+        ImageLoadingURLProtocolSpy.reset()
+        wayNameView.label.spriteRepository.resetCache()
+        wayNameView.label.representation = nil
+    }
+
+    override func tearDown() {
+        super.tearDown()
+    }
+    
+    func loadingURL(baseURL: URL, styleID: String) {
+        guard let spriteRequestURL = wayNameView.label.spriteRepository.spriteURL(isImage: true, baseURL: baseURL, styleID: styleID),
+              let metadataRequestURL = wayNameView.label.spriteRepository.spriteURL(isImage: false, baseURL: baseURL, styleID: styleID) else {
+                  XCTFail("Failed to form request to update SpriteRepository")
+                  return
+              }
+        
+        let scale = Int(VisualInstruction.Component.scale)
+        ImageLoadingURLProtocolSpy.registerData(ShieldImage.shield.image.pngData()!, forURL: spriteRequestURL)
+        ImageLoadingURLProtocolSpy.registerData(Fixture.JSONFromFileNamed(name: "sprite-info"), forURL: metadataRequestURL)
+        ImageLoadingURLProtocolSpy.registerData(ShieldImage.shield.image.pngData()!, forURL:  URL(string: ShieldImage.i280.baseURL.absoluteString + "@\(scale)x.png")!)
+    }
+    
+    func testUpdateStyle() {
+        let baseURL = wayNameView.label.spriteRepository.baseURL
+        let styleID = "/mapbox/navigation-night-v1"
+        loadingURL(baseURL: baseURL, styleID: styleID)
+        
+        let shield = VisualInstruction.Component.ShieldRepresentation(baseURL: baseURL, name: "us-interstate", textColor: "white", text: "280")
```

Comment:
The `za-national` (South African national route) shield type has yellow text, which would make for a more interesting test of our color conversion capabilities.